src/ui/common/src/reducers/integration.ts

```diff
@@ -154,7 +154,17 @@ export const handleLoadIntegrationObject = createAsyncThunk<
       return thunkAPI.rejectWithValue(objectResponseBody.error);
     } else {
       const serialized = (objectResponseBody as ObjectPreviewResponse).data;
-      return JSON.parse(serialized) as Data;
+      const rawData = JSON.parse(serialized);
+      if ('schema' in rawData) {
```

Comment:
Seems that here, we are using whether `schema` exist to identify whether it's a bson table, but in the code above we are explicitly checking the serialization type. Is this because here it's hard to get access to that property?